app/javascript/vue/shared/select_dropdown.vue

```diff
@@ -354,10 +367,15 @@ export default {
     },
     fetchOptions() {
       if (this.optionsUrl) {
-        const params = { query: this.query, ...this.urlParams };
+        const params = { query: this.query, page: this.nextPage, ...this.urlParams };
         axios.get(this.optionsUrl, { params })
           .then((response) => {
-            this.fetchedOptions = response.data.data;
+            if (response.data.paginated) {
```

Comment:
I think we could just check response.data.next_page here, no need for the `paginated` flag